lua/controller.lua

```diff
@@ -46,6 +46,30 @@ local pipeworks_loaded = core.get_modpath("pipeworks") and pipeworks
 local digilines_loaded = core.get_modpath("digilines") and digilines
 local techage_loaded = core.get_modpath("techage") and techage
 
+local max_matches = tonumber(core.settings:get("drawers.controller_max_matches")) or 50
+
+-- Cache content IDS of all registered drawer items
+local controller_content_id
+local trim_content_id = core.get_content_id("drawers:trim")
+local drawer_content_ids = {}
+for name, _ in pairs(core.registered_items) do
+	if core.get_item_group(name, "drawer") > 0 or core.get_item_group(name, "drawer_connector") > 0 then
+		drawer_content_ids[core.get_content_id(name)] = true
+	end
+end
```

Comment:
This should run in `on_mods_loaded`, in case further registrations follow after this code is executed.